internal/xds/translator/cluster.go

```diff
@@ -858,26 +851,34 @@ func buildTypedExtensionProtocolOptions(args *xdsClusterArgs) (map[string]*anypb
 	case args.useClientProtocol:
 		protocolOptions.UpstreamProtocolOptions = &httpv3.HttpProtocolOptions_UseDownstreamProtocolConfig{
 			UseDownstreamProtocolConfig: &httpv3.HttpProtocolOptions_UseDownstreamHttpConfig{
-				HttpProtocolOptions:  http1opts,
-				Http2ProtocolOptions: buildHTTP2Settings(args.http2Settings),
+				HttpProtocolOptions:  http1Opts,
+				Http2ProtocolOptions: http2Opts,
 			},
 		}
 	case requiresHTTP2Options:
 		protocolOptions.UpstreamProtocolOptions = &httpv3.HttpProtocolOptions_ExplicitHttpConfig_{
 			ExplicitHttpConfig: &httpv3.HttpProtocolOptions_ExplicitHttpConfig{
 				ProtocolConfig: &httpv3.HttpProtocolOptions_ExplicitHttpConfig_Http2ProtocolOptions{
-					Http2ProtocolOptions: buildHTTP2Settings(args.http2Settings),
+					Http2ProtocolOptions: http2Opts,
 				},
 			},
 		}
 	case requiresHTTP1Options:
 		protocolOptions.UpstreamProtocolOptions = &httpv3.HttpProtocolOptions_ExplicitHttpConfig_{
 			ExplicitHttpConfig: &httpv3.HttpProtocolOptions_ExplicitHttpConfig{
 				ProtocolConfig: &httpv3.HttpProtocolOptions_ExplicitHttpConfig_HttpProtocolOptions{
-					HttpProtocolOptions: http1opts,
+					HttpProtocolOptions: http1Opts,
 				},
 			},
 		}
+	case requiresAutoHTTPConfig:
+		// use Auto when there's a transport socket
+		protocolOptions.UpstreamProtocolOptions = &httpv3.HttpProtocolOptions_AutoConfig{
+			AutoConfig: &httpv3.HttpProtocolOptions_AutoHttpConfig{
```

Comment:
should auto be used for default as well ?

Reply:
this's why I limit the scope in this PR, we probably want to cherrypick this one to `v1.5`.